src/toplc.ml

```diff
@@ -656,17 +651,92 @@ let get_ancestors h c =
   ga c;
   U.hashtbl_fold_keys (fun c cs -> c :: cs) cs []
 
-let name_of_class c =
-  B.Utils.UTF8.to_string (B.Name.external_utf8_for_class c)
-
-let mk_full_method_name c mn =
-  name_of_class c ^ "." ^ mn
-
 let get_overrides h c m =
   let ancestors = get_ancestors h c in
   let qualify c =  mk_full_method_name c m.method_name in
   (List.map qualify ancestors, m.method_arity)
 
+let rec pp_for_parameter_list = function
+  | [] -> ""
+  | p :: [] -> B.Utils.UTF8.to_string(BD.utf8_of_parameter(p))
+  | p :: ps -> B.Utils.UTF8.to_string(BD.utf8_of_parameter(p)) ^ pp_for_parameter_list(ps)
+
+let if_ b xs = if b then List.concat xs else []
+
+let has = (<>) None
+        
+let instrument_call locals parameters call_id instrs =
+  let rev_params = List.rev parameters in
+  let max = locals + List.length parameters in
+  let instrumentation =
+    if log_in then printf "Call Instrumented with call ID: %a, Locals: %i, Max: %i, Parameter List: %s\n"
+                   (U.pp_option U.pp_int) call_id locals max (pp_for_parameter_list parameters); 
```

Comment:
At some point we should migrate from `Util.pp_option` to `Format.pp_print_option`, which was added in OCaml 4.08.

Reply:
I have migrated over to `Format.pp_print_option`